pkg/api/upgrade.go

```diff
@@ -23,18 +24,34 @@ type UpgradeResponse struct {
 func Upgrade(svc Service) http.Handler {
 	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
 		defer r.Body.Close()
-
 		var req UpgradeRequest
 
 		if err := json.NewDecoder(r.Body).Decode(&req); err == io.EOF || err != nil {
-			logger.Errorf("[Upgrade] error decoding request: %v", err.Error())
 			w.WriteHeader(http.StatusBadRequest)
+			logger.Errorf("[Upgrade] error decoding request: %v", err)
 			return
 		}
-
 		defer r.Body.Close()
+
+		var install bool
+		var version string
+		for key, value := range req.Flags {
```

Comment:
why do we've hardcoded flags? This means we only support install and version, it's not generic as well.

later, we'll extend it general for flags and set flags in code as it's used in helm code flow.

`value/options.go`  `MergeValues` seems to be handling it.